internal/scm/bitbucketcloud/bitbucket_cloud.go

```diff
@@ -0,0 +1,324 @@
+		}),
+	}
+	bitbucketCloud.bbClient = bitbucket.NewBasicAuth(username, token)
+
+	return bitbucketCloud, nil
+}
+
+func (bbc *BitbucketCloud) CreatePullRequest(ctx context.Context, repo scm.Repository, prRepo scm.Repository, newPR scm.NewPullRequest) (scm.PullRequest, error) {
+
+	splitRepoFullName := strings.Split(prRepo.FullName(), "/")
+
+	repoOptions := &bitbucket.RepositoryOptions{
+		Owner:    bbc.workspaces[0],
+		RepoSlug: splitRepoFullName[1],
+	}
+	defaultReviewers, _ := bbc.bbClient.Repositories.Repository.ListDefaultReviewers(repoOptions)
+	for _, reviewer := range defaultReviewers.DefaultReviewers {
+		newPR.Reviewers = append(newPR.Reviewers, reviewer.Uuid)
+	}
+
+	prOptions := &bitbucket.PullRequestsOptions{
+		Owner:             bbc.workspaces[0],
+		RepoSlug:          splitRepoFullName[1],
+		SourceBranch:      newPR.Head,
+		DestinationBranch: newPR.Base,
+		Title:             newPR.Title,
+		CloseSourceBranch: true,
+		Reviewers:         newPR.Reviewers,
+	}
+
+	_, err := bbc.bbClient.Repositories.PullRequests.Create(prOptions)
+	if err != nil {
+		return nil, err
+	}
+
+	return &pullRequest{
+		project:    splitRepoFullName[0],
+		repoName:   splitRepoFullName[1],
+		branchName: newPR.Head,
+		prProject:  splitRepoFullName[0],
+		prRepoName: splitRepoFullName[1],
+		status:     scm.PullRequestStatusSuccess,
+	}, nil
+}
+
+func (bbc *BitbucketCloud) UpdatePullRequest(ctx context.Context, repo scm.Repository, pullReq scm.PullRequest, updatedPR scm.NewPullRequest) (scm.PullRequest, error) {
+	bbcPR := pullReq.(pullRequest)
+	repoSlug := strings.Split(bbcPR.guiURL, "/")
+
+	// Note the specs of the bitbucket client here, reviewers field must be UUID of the reviewers, not their usernames
+	prOptions := &bitbucket.PullRequestsOptions{
+		ID:                fmt.Sprintf("%d", bbcPR.number),
+		Owner:             bbc.workspaces[0],
+		RepoSlug:          repoSlug[4],
+		Title:             updatedPR.Title,
+		Description:       updatedPR.Body,
+		CloseSourceBranch: true,
+		SourceBranch:      updatedPR.Head,
+		DestinationBranch: updatedPR.Base,
+		Reviewers:         updatedPR.Reviewers,
+	}
+	_, err := bbc.bbClient.Repositories.PullRequests.Update(prOptions)
+	if err != nil {
+		return nil, err
+	}
+
+	return &pullRequest{
+		project:    bbcPR.project,
+		repoName:   bbcPR.repoName,
+		branchName: updatedPR.Head,
+		prProject:  bbcPR.prProject,
+		prRepoName: bbcPR.prRepoName,
+		status:     scm.PullRequestStatusSuccess,
+	}, nil
+}
+
+func (bbc *BitbucketCloud) GetPullRequests(ctx context.Context, branchName string) ([]scm.PullRequest, error) {
+	var responsePRs []scm.PullRequest
+	for _, repoName := range bbc.repositories {
+		prs, _ := bbc.bbClient.Repositories.PullRequests.Gets(&bitbucket.PullRequestsOptions{Owner: bbc.workspaces[0], RepoSlug: repoName})
+		prBytes, _ := json.Marshal(prs)
+		bbPullRequests := &bitbucketPullRequests{}
+		err := json.Unmarshal(prBytes, bbPullRequests)
+		if err != nil {
```

Comment:
I find it strange that the bbc package does return `any` (`interface{}`) and not concrete types. That should be one of the biggest benefits of using a pre defined package instead of rolling your own. But let us work under this constraint. 

Since we have to do the marshal->unmarshal in multiple places, there would be benefit in creating a helper function for this

```go
func decode(raw any, structured any) error {
	prBytes, err := json.Marshal(prs)
    if err != nil { // I guess this should never happen, since reference loops etc should not exist, but does not hurt to check it
        return err 
    }
    return json.Unmarshal(prBytes, structured)
}
```

An alternative is to use https://github.com/mitchellh/mapstructure , but I'm fine if you want to do go the json approach since that is where it is unmarshaled from to begin with.

Reply:
We only had to do the marshal/unmarshal a few times so I decided not to add the decode function, I don't think its excessively repetitive to keep it as is but let me know if you think its blocking